### PR TITLE
Fix 18121: Add .pxd linting to lint.sh

### LINKED ISSUE #18121: CI: add .pxd linting to lint.sh

### PR DESCRIPTION
**Pull request by manrajgrover**
- [ ] closes #18121 
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: ci/lint.sh [hunk 1]
```diff
@@ -42,6 +42,18 @@ if [ "$LINT" ]; then
     done
     echo "Linting *.pxi.in DONE"
 
+    echo "Linting *.pxd"
+    for path in 'src'
```
Comment (jreback) [inline ci/lint.sh]: this should recursively search _libs

Comment (jreback) [reply ci/lint.sh]: i think there may be some fixes needed as well to make this pass (to the actual .pxd files)

**Review by jreback — CHANGES_REQUESTED** [review]

File: ci/lint.sh (hunk 1 above)
Comment (ManrajGrover) [reply ci/lint.sh]: @jreback Got it. Is there a way to make `lint.sh` run locally?

Comment (jreback) [reply ci/lint.sh]: not really but you can just check the command locally

Comment (jbrockmendel) [reply ci/lint.sh]: >  Is there a way to make lint.sh run locally?

On OSX/Linux:
```
export LINT=TRUE; /bin/bash ci/lint.sh
```

The first line of output is an error message you can ignore.

**Comment by jreback** [comment]
rebase on master

File: pandas/_libs/src/numpy.pxd [hunk 2]
```diff
@@ -598,8 +603,10 @@ cdef extern from "numpy/arrayobject.h":
     object PyArray_Dumps (object, int)
     int PyArray_ValidType (int)
     void PyArray_UpdateFlags (ndarray, int)
-    object PyArray_New (type, int, npy_intp *, int, npy_intp *, void *, int, int, object)
-    #object PyArray_NewFromDescr (type, dtype, int, npy_intp *, npy_intp *, void *, int, object)
+    object PyArray_New (type, int, npy_intp *, int, npy_intp *,
+                        void *, int, int, object)
+    #object PyArray_NewFromDescr \
```
Comment (jreback) [inline pandas/_libs/src/numpy.pxd]: you can't use ``\`` here, this is kind of like a c-header file. you can just remove this line as its not used anyhow (its commented out)

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @ManrajGrover

**Pull request merged**